### ISSUE TITLE
Switch to Clipboard API

### ISSUE THREAD

**Issue by sonicdoe** [sindresorhus/copy-text-to-clipboard#35]
Chrome, Edge, Firefox, and Safari ([“modern browsers”](https://github.com/sindresorhus/copy-text-to-clipboard/issues/6#issuecomment-306801161)) now support the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API), allowing you to copy text to the clipboard using [`navigator.clipboard.writeText()`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText).

Should this package switch to the Clipboard API (or be deprecated in favor of it)?

**Comment by sindresorhus** [comment]
That API is new and probably buggy, and it's async, which can be annoying sometimes. I'll leave this issue open for now, but I don't intend to do anything about it yet.

**Comment by andrewmclagan** [comment]
Would you consider a BC and major version bump?

**Comment by Richienb** [comment]
The async clipboard API is now supported in [`clipboardy`](https://github.com/sindresorhus/clipboardy).

**Comment by jamiehaywood** [comment]
MDN now says that `execCommand` [is deprecated](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard)

**Comment by sindresorhus** [comment]
Deprecated just means discouraged on the web. Nothing can ever be removed.

**Comment by fregante** [comment]
To anyone reaching this issue: If you can use the new API, use it directly. This module was good when the solution is complex. Using the new API here would just mean:

```js
const copy = (text) => navigator.clipboard.writeText(text)
```

It doesn't make sense to add a dependency anymore. However the new API seems to be slightly limited compared to `execCommand` so you might encounter this error if used in a frame (e.g. Chrome extensions devtools):


> pageEditor.js:198997 Uncaught (in promise) DOMException: The Clipboard API has been blocked because of a permissions policy applied to the current document. See https://goo.gl/EuHzyv for more details.

**Comment by Matt-Jensen** [comment]
Rather than switching to the Clipboard API, its more important to update the README to recommend the Clipboard API instead of installing this package.

This package can continue to be useful for older browsers, however we should be encouraging users to **not install this package** if their [target environments](https://caniuse.com/mdn-api_clipboard_writetext) are up to snuff.  Probably change all wording related to "modern browsers" as well.

**Comment by silverwind** [comment]
> Rather than switching to the Clipboard API, its more important to update the README to recommend the Clipboard API instead of installing this package.

The old method is still useful as a fallback when running on insecure contexts. It's what I do in [clippie](https://github.com/silverwind/clippie).

**Comment by tomleo** [comment]
Yeah this package is super misleading, this isn't a solution for modern browsers, it's a solution that's deprecated (`execCommand`). I would just stop branding this as a solution for "modern browsers".

> That API is new and probably buggy

It's not.

> and it's async, which can be annoying sometimes. 

If by sometimes you mean IO in node, then I guess sometimes, but in a modern browser it's not.
